lib/mongo/server/connection_pool/queue.rb

```diff
@@ -88,7 +87,7 @@ def disconnect!
         # @since 2.0.0
         def enqueue(connection)
           mutex.synchronize do
-            queue.unshift(connection)
+            queue.unshift(connection.record_checkin!)
             resource.broadcast
           end
         end
```

Comment:
No. ``create_connection`` calls these methods:
https://github.com/estolfo/mongo-ruby-driver/blob/master/lib/mongo/cluster.rb#L269
https://github.com/estolfo/mongo-ruby-driver/blob/master/lib/mongo/server/connection_pool.rb#L132

which eventually creates a Connection, with ``last_checkin`` initialized to nil [here](https://github.com/estolfo/mongo-ruby-driver/blob/RUBY-741-max-idle-time/lib/mongo/server/connection.rb#L142).

Also, ``last_checkin`` is set to nil when the connection is [disconnected](https://github.com/estolfo/mongo-ruby-driver/blob/RUBY-741-max-idle-time/lib/mongo/server/connection.rb#L91).